### PR TITLE
[DOCS][Transform] document limitation regarding rolling upgrade with 7.2, 7.3

### PR DESCRIPTION
**Pull request by hendrikmuhs**
adds a limitation about rolling upgrade, see #48019

also fixes a problem with renamed preferences

Preview: http://elasticsearch_48118.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-limitations.html

### CODE REVIEW

File: docs/reference/transform/limitations.asciidoc [hunk 1]
```diff
@@ -33,6 +33,14 @@ upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2,
 the {transforms} UI (earler {dataframe} UI) will not work. Please wait until all 
 nodes have been upgraded to the newer version before using the {transforms} UI.
 
+[float]
+[[transform-rolling-upgrade-limitation]]
+==== {transforms-cap} Transforms suspended during a rolling upgrade from 7.2 and 7.3
+
+If your cluster contains mixed version nodes, for example during a rolling
+upgrade from 7.2, 7.3 to a newer version, transforms are suspended for the time of
```
Comment (droberts195) [inline docs/reference/transform/limitations.asciidoc]: Is "suspended" the right word here?  I thought that we would not assign transform tasks in a mixed version cluster that includes pre-7.4 nodes, but we will not actually _suspend_ transforms that are already running before the upgrade starts.

If so the the issue title could be "Transforms reassignment suspended during a rolling upgrade from 7.2 and 7.3", and "transforms are suspended for the time of the upgrade" could be "transforms whose nodes are stopped will not be reassigned until the upgrade is complete".

File: docs/reference/transform/limitations.asciidoc [hunk 2]
```diff
@@ -33,6 +33,14 @@ upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2,
 the {transforms} UI (earler {dataframe} UI) will not work. Please wait until all 
 nodes have been upgraded to the newer version before using the {transforms} UI.
 
+[float]
+[[transform-rolling-upgrade-limitation]]
+==== {transforms-cap} Transforms reassignment suspended during a rolling upgrade from 7.2 and 7.3
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]:
```suggestion
==== {transforms-cap} reassignment suspended during a rolling upgrade from 7.2 and 7.3
```

File: docs/reference/transform/limitations.asciidoc [hunk 3]
```diff
@@ -33,6 +33,14 @@ upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2,
...
+
+If your cluster contains mixed version nodes, for example during a rolling
+upgrade from 7.2, 7.3 to a newer version, transforms whose nodes are stopped will
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]:
```suggestion
upgrade from 7.2 and 7.3 to a newer version, transforms whose nodes are stopped will
```

File: docs/reference/transform/limitations.asciidoc [hunk 4]
```diff
@@ -33,6 +33,14 @@ upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2,
...
+not be reassigned until the upgrade is complete. Once the upgrade is done, transforms
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]:
```suggestion
not be reassigned until the upgrade is complete. After the upgrade is done, {transforms}
```

File: docs/reference/transform/limitations.asciidoc [hunk 5]
```diff
@@ -33,6 +33,14 @@ upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2,
...
+will resume automatically, there is no action required.
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]:
```suggestion
resume automatically; no action is required.
```

File: docs/reference/transform/limitations.asciidoc (hunk 3 above)
Comment (hendrikmuhs) [reply docs/reference/transform/limitations.asciidoc]: In a rolling upgrade there are only 2 versions involved, "from" is either 7.2 _or_ 7.3 and "to" is 7.4, 7.5, 7.6...

Sounds to me like "or" would be the right choice.

**Review by szabosteve — APPROVED** [review]
LGTM

**Pull request merged**